docs/release-notes/self-hosted/self-hosted-v4.0.0.md

```diff
@@ -0,0 +1,84 @@
+# Self-hosted v4.0.0
+
+These release notes are for [Codacy Self-hosted v4.0.0](https://github.com/codacy/chart/releases/tag/4.0.0){: target="_blank"}, released on May 12, 2021. <!-- TODO Update release date -->
+
+To upgrade Codacy, follow [these instructions](../../chart/maintenance/upgrade.md).
+
+<!-- TODO Check these issues
+
+Check the following Jira Bugs with disabled release notes:
+
+-   https://codacy.atlassian.net/browse/CY-4161
+
+-->
+
+## Breaking changes
+
+This version of Codacy Self-hosted requires [PostgreSQL version 10.15](https://docs.codacy.com/v4.0/chart/requirements/#postgresql-server-setup) since the previous major version [will reach end-of-life (EOL) later this year](https://www.postgresql.org/support/versioning/).
+
+To upgrade PostgreSQL:
+
+1.  Follow the instructions on [how to upgrade a PostgreSQL cluster](https://www.postgresql.org/docs/10/upgrading.html).
+
+1.  After the upgrade, clean the Codacy databases with [vacuumdb](https://www.postgresql.org/docs/10/app-vacuumdb.html) to prevent the <span class="skip-vale">autovacuum</a> feature from momentarily affecting the performance of the databases at a later time.
+
+## Product enhancements
+
+-   Improved the loading time of the [repositories list](https://docs.codacy.com/v4.0/organizations/managing-repositories/) page when an organization has many repositories. (CY-4084)
+-   Now, Codacy allows [configuring who is allowed to manage file extensions and branches](https://docs.codacy.com/v4.0/organizations/roles-and-permissions-for-synced-organizations/#change-analysis-configuration). (CY-3886, CY-3887)
+
+## Bug fixes
+
+-   Fixed wrong feedback message that mentioned disabling tools when toggling between the UI settings and configuration file for a tool on the Code patterns page. (CY-4276)
+-   Fixed an issue where Codacy could fail to show repositories for organizations with more than 1000 repositories on GitHub. (CY-4193)
+-   Fixed an issue that caused some cookies to be transmitted in plaintext when accessing `app.codacy.com` via HTTP with Firefox. (CY-4143)
```

Comment:
@caxaria @h314to What's our policy regarding [vulnerability disclosure](https://en.wikipedia.org/wiki/Full_disclosure_(computer_security))?

Should we:

- Announce this fix on the release notes for this release?
- Mention that the release includes security fixes but without providing the exact details of the issue?

Reply:
Seems ok 👍